replica/table.cc

```diff
@@ -1216,6 +1217,21 @@ table::make_memtable_list() {
     return make_lw_shared<memtable_list>(std::move(seal), std::move(get_schema), _config.dirty_memory_manager, _stats, _config.memory_compaction_scheduling_group);
 }
 
+compaction_group::compaction_group(table& t)
+    : _t(t)
+    , _table_state(std::make_unique<table_state>(t))
+{
+    _t._compaction_manager.add(as_table_state());
+}
+
+future<> compaction_group::stop() noexcept {
+    try {
+        return _t._compaction_manager.remove(as_table_state());
```

Comment:
Better to change compaction_manager::remove to be noexcept.

Reply:
Will do.